modules/openapi-generator/src/main/java/org/openapitools/codegen/config/CodegenConfigurator.java

```diff
@@ -480,6 +481,13 @@ public CodegenConfigurator setPackageName(String packageName) {
         return this;
     }
 
+    public CodegenConfigurator setResourceFolder(String resourceFolder) {
+        if (StringUtils.isNotEmpty(resourceFolder)) {
+            addAdditionalProperty(SpringCodegen.RESOURCE_FOLDER, resourceFolder);
+        }
+        return this;
+    }
+
```

Comment:
I don't think there should be change to this file as resourceFolder is not a global option.

It's a generator option, which should be passed via `additionalProperties`